### PR TITLE
WEB-8791: Create input components

### PR DESCRIPTION
**Pull request by davidicus**
Closes WEB-8791

**Summary**

- This PR introduces two new components that take care of our form input needs. 

**Change List (describe the changes made to the files)**

- Added an `Input.tsx` file for the base component
- Added `Index.test.tsx' to test
- Added `index.scss` file for styles
- Updated the index exports file to include this new component
- Updated the storybook config to deal with a storybook autodocs issue
- Updated vite config for alias for utils file
- Added a input normalizer function to utils folder
- removed the storybook version of testing library as it was clashing and causing some act errors
- removed instances where we were using the storybook testing utils
- updated the typography partial 
- added hidden mixin to the utils partial 
- created the variables for the distinct font that were being used in the mixins
- Added input partial to the main sass file
- Added a toggle partial and story.

**Acceptance Test (how to verify the PR)**

- play around with each input type and make sure it functions normally

**Things to look for during review**

- [ ] Make sure all references to `phillips` class prefix is using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] All strings should be translatable.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue
- [ ] Make sure all commits messages follow convention and are appropriate for the changes

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default {
     "^.+\\.tsx?$": "ts-jest"
   },
   collectCoverage: true,
-  collectCoverageFrom: ['src/components/**/*.ts?(x)', '!src/**/*.stories.ts?(x)',],
+  collectCoverageFrom: ['src/components/**/*.ts?(x)', '!src/**/*.stories.ts?(x)', 'src/utils/*.ts?(x)'],
```
Comment (davidicus) [inline jest.config.ts:10]: Added a utils file to the list of files to be tested in jest config.

File: package.json [hunk 2]
```diff
@@ -47,10 +47,9 @@
     "@storybook/blocks": "^7.0.22",
     "@storybook/react": "^7.0.22",
     "@storybook/react-vite": "^7.0.22",
-    "@storybook/testing-library": "^0.0.14-next.2",
```
Comment (davidicus) [inline package.json:50]: I removed the storybook version as it was clashing with the version we use for testing. Updated the user even library.

File: src/pages/Page.stories.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import type { Meta, StoryObj } from '@storybook/react';
-import { within, userEvent } from '@storybook/testing-library';
+// import { within, userEvent } from '@storybook/testing-library';
```
Comment (davidicus) [inline src/pages/Page.stories.ts:2]: removed references to the old storybook. This page will be rewritten and is only here for proof of concept.

File: tsconfig.json [hunk 4]
```diff
@@ -9,6 +9,7 @@
       "./vars": ["./src/scss/vars"],
       "./typography": ["./src/scss/typography"],
       "./reset": ["./src/scss/reset"],
+      "./utils": ["./src/scss/utils"],
```
Comment (davidicus) [inline tsconfig.json:12]: mapped alias for build

File: vite.config.ts [hunk 5]
```diff
@@ -36,6 +36,14 @@ export default defineConfig({
...
         replacement: resolve(__dirname, './src/scss/reset')
       },
+      {
+        find: './_utils',
+        replacement: resolve(__dirname, './src/scss/utils')
+      },
+      {
+        find: '../../_utils',
+        replacement: resolve(__dirname, './src/scss/utils')
+      },
```
Comment (davidicus) [inline vite.config.ts:46]: mapped alias for build

File: src/pages/Page.stories.ts (hunk 3 above)
Comment (alexandertumbasov) [reply src/pages/Page.stories.ts:2]: Hey @storybook/testing-library will be useful in the later stages of a project without a lot of technical debt. I suggest removing this library completely and using those libraries for unit tests
@testing-library/user-event
@testing-library/react

Comments need to be deleted

File: src/components/Select/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+export interface SelectProps extends InputProps {
+  /**
+   * Option elements that are selectable
+   */
+  children: React.ReactNode;
+
+}
+
+const Select = React.forwardRef((
```
Comment (alexandertumbasov) [inline src/components/Select/Select.tsx:16]: import {forwardRef} from 'react'

File: src/components/Input/Input.test.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    expect(screen.queryByText(/invalid/)).not.toBeInTheDocument();
+  });
+
+  it('will fire the onClick handler when clicked', async () => {
+    const mockedOnClick = jest.fn();
+    render(<Input {...reqProps} onClick={mockedOnClick} />)
+    userEvent.click(screen.getByTestId('test-id'));
+    await waitFor(() => expect(mockedOnClick.mock.calls).toHaveLength(1));
+  });
+});
```
Comment (Sisawat) [inline src/components/Input/Input.test.tsx:103]: I know it doesn't really matter, but trailing linefeeds should be present to suppress github complaining about this. Does VSCode not automatically add them by filetype?

**Review by Sisawat — APPROVED** [review]

File: src/components/Input/Input.test.tsx (hunk 7 above)
Comment (davidicus) [reply src/components/Input/Input.test.tsx:103]: It should once I install prettier.

File: src/components/Select/Select.tsx (hunk 6 above)
Comment (davidicus) [reply src/components/Select/Select.tsx:16]: It is a convention. I prefer verbosity to explicitly show what is coming from  React. It also does not require me to continuously update my import statement if and when I need to pull another method from React.

File: src/pages/Page.stories.ts (hunk 3 above)
Comment (davidicus) [reply src/pages/Page.stories.ts:2]: As I said, this page is only here for conceptual purposes and will eventually be totally overwritten. I appreciate the input on  the storybook testing-library but it clashed with existing testing framework and until I have had the time to look into storybooks testing capabilities and how it affects our plans for our CI/CD pipeline I am removing.

**Pull request merged**